cql3/untyped_result_set.cc

```diff
@@ -30,17 +30,17 @@ size_t cql3::untyped_result_set_row::index(const std::string_view& name) const {
 bool cql3::untyped_result_set_row::has(std::string_view name) const {
     auto i = index(name);
     if (i < _data.size()) {
-        return !std::holds_alternative<std::monostate>(_data.at(i));
+        return _data.at(i).has_value();
```

Comment:
A note about the commit message (which github doesn't give me a better place to put):
Your observations about untyped_result_set being slow-path, used only for internal queries, and (after this patch) owning, are all important observations that another developer is unlikely to guess in the future and may easily break by mistake in new code.  I think we should have a comment in untyped_result_set.hh saying those things.

Reply:
I'll document it (in a separate pull request).

Reply:
https://github.com/scylladb/scylladb/pull/13661